### PR TITLE
[pulsar-io] Add NSQ Source

### PR DESCRIPTION
**Pull request by flowchartsman**
### Motivation

This PR adds a source representing an [NSQ](https://nsq.io/) topic to be mirrored to a pulsar topic.

### Modifications

With the exception of adding it as a module to `pulsar-io/pom.xml`, there are no modifications to existing code.

### Verifying this change

I need some help with this, since it is adapted from a local, standalone repo, and I am having difficulties getting maven to resolve dependencies for other parts of the Pulsar codebase, which prevents me from testing it properly in situ.

This change added tests and can be verified as follows:

This change is largely based off of the Twitter Firehose example, which tests only its config, and I have written similar tests, as well as an extra for the default channel option.  Any guidance on further tests is appreciated.

### Does this pull request potentially affect one of the following parts:
N/Z

### Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (following the twitter example, it is documented basically in code)

Note: if I have missed any other links to documentation, please mention them in the comments and I will add them.

### CODE REVIEW

**Comment by flowchartsman** [comment]
Turns out I was building against JDK 14. Used Jenv to install JDK 11, correcting a couple of errors and will issue another push.

**Comment by flowchartsman** [comment]
I believe everything should be in order now. I am unsure why these checks are failing, but they don't seem to be related. Awaiting review.

**Comment by codelipenghui** [comment]
@flowchartsman Thanks for the great contribution, we will review soon.

File: site2/docs/io-nsq-source.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+The configuration of the NSQ source connector has the following properties.
+
+### Property
+
+| Name | Type|Required | Default | Description 
+|------|----------|----------|---------|-------------|
+| `lookupds` |String| true | " " (empty string) | A comma-separated list nsqlookupds to connect to. |
+| `topic` | String|true | " " (empty string) | The NSQ topic to transport |
+| `channel` | String |false | pulsar-transport-{$topic} | The channel to register and consume. from on the provided NSQ topic. |
```
Comment (Huanli-Meng) [inline site2/docs/io-nsq-source.md:20]: The sentence in the "description" column should be updated from the grammatical perspective.

Comment (flowchartsman) [reply site2/docs/io-nsq-source.md]: Thank you! I have altered the descriptions for `lookupds` and `channel`

**Comment by sijie** [comment]
@codelipenghui Can you review this?

**Comment by flowchartsman** [comment]
Pending the review, now is probably a good time to ask about when to add any knobs that the user might want.  There are certainly a bunch of NSQ configuration options that could be supplied, this is a pretty bare bones transporter. I am happy to add a few now, though some of them (including auth) would rely on secrets, which I might need some help implementing.

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@flowchartsman There are some flaky test that already fixed in master branch, could you please sync with the apache master
branch? Since you are pushing the PR with your master branch, I have sync the apache master branch to your master branch.

**Comment by flowchartsman** [comment]
I believe I may have issued a rebase incorrectly, I will attempt to resolve

**Review by sijie — APPROVED** [review]

**Comment by flowchartsman** [comment]
@codelipenghui I see some tests still failing—are we good, or do I need to make some more changes?

**Pull request merged**